Azure/iot-identity-service
# Change aziot-keys to use pkcs11 directly instead of through pkcs11-openssl-engine.

The PKCS#11 openssl engine made sense before the final design that separated
KS, IS and MR, since the expectation was that iotedged would use its EVP_PKEYs
directly.

Now that PKCS#11 is only used through aziot-keys, converting from aziot-keys's
API to openssl's engine API and then back to PKCS#11 API is overkill.

The only other use of pkcs11-openssl-engine was for pkcs11-test. Rather than
delete pkcs11-test too, this commit repurposes it to test
the aziot_keys openssl engine. Documentation has been updated accordingly.

Other changes:

- Fix missing phony `test` target. 47e14a7a60026a81173f54dbb23b2974bd759db3
  renamed it to `test-release` instead of appending `test-release` to the list.

- Fix e2e-tests.sh to exit quickly if the `AZURE_*` env vars are not set.
  5d5bf7b110b2d177c27516ec3f2245d586096735 tried to do this by asserting that
  the variables were defined, but that check was insufficient. When the secrets
  aren't defined in GitHub, the variables are defined and empty.

- Consistently use `;;` after `exit` in `case` arms.

- Remove P-384, P-521 and ED25519-related code from the PKCS#11 code. It was
  unused since we only support P-256 through aziot-keys.

- Remove disabled `arm32v7` and `aarch64` stubs from the GH checkin workflow.
